main.go

```diff
@@ -1,7 +1,190 @@
 package main
 
-import "fmt"
+
+import (
+
+    "net/http"
+
+ 
+
+    "github.com/gin-gonic/gin"
+
+)
+
+ 
+
+type NameList struct {
+
+    NameF string  `json: "fName:"`
+
+    NameL string  `json: "lName:"`
+
+    Age   float32 `json: "Age:"`
+
+}
+
+ 
+
+var names = []NameList{
+
+    {NameF: "Quandale", NameL: "Dingle the third", Age: 59},
+
+    {NameF: "James", NameL: "Howlet", Age: 134},
+
+    {NameF: "Bihan", NameL: "Tundra", Age: 38},
+
+}
+
+ 
+
+type Account struct{
+
+    Username string `json:"username"`
+
+    Password string `json:"password"`
+
+    Role string `json:"role"`
+
+}
+
+ 
+
+var accounts []Account
+
+func Register(c *gin.Context){
+
+    var account Account
+
+    if  err := c.ShouldBindJSON(&account); err != nil {
+
+        c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+
+        return
+
+    }
+
+    accounts = append(accounts, account)
+
+    c.JSON(http.StatusOK, gin.H{"message": "Account registered succesfully ",
+
+"account": account,
+
+})
+
+}
+
+ 
+
+func getNames(c *gin.Context) {
+
+    c.IndentedJSON(http.StatusOK, names)
+
+}
+
+ 
+
+func handlerFunc(c *gin.Context) {
+
+    c.String(200, "My record of Kira's kills")
+
+}
+
+ 
 
 func main() {
-  fmt.Println("hello world")
+
+ 
+
+    router := gin.Default()
+
+    router.GET("/", handlerFunc)
+
+    router.GET("/names", getNames)
+
+    router.PUT("/names", putNames)
+
+    router.POST("/names", postNames)
+
+    router.DELETE("/names", deleteNames)
+
+    router.POST("/register", Register)
+
+    router.Run()
+
+}
+
+func deleteNames(C *gin.Context) {
+
+    var deleteItem NameList
+
+    if err := C.ShouldBindJSON(&deleteItem); err != nil {
+
+        C.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+
+        return
+
+    }
+
```

Comment:
create a separate handler package and a separate models package. This is where u can put your handler functions and your struct definitions respectively and import them into main. This makes ur code more organized and cleaner.